pkg/reconciler/instances/istio/istioctl/resolver.go

```diff
@@ -0,0 +1,152 @@
+package istioctl
+
+import (
+	"fmt"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/pkg/errors"
+)
+
+const (
+	versionSuffixSeparator = "-"
+)
+
+type Version struct {
+	value semver.Version
+}
+
+//Returns a Version from passed semantic version in the format: "major.minor.patch", where all components must be positive integers
```

Comment:
Docs should start with the name of the function.

Reply:
Done.